narwhals/_pandas_like/dataframe.py

```diff
@@ -1198,7 +1198,8 @@ def explode(self, columns: Sequence[str]) -> Self:
 
         exploded_frame = native_frame[[*other_columns, columns[0]]].explode(columns[0])
         exploded_series = [
-            native_frame[col_name].explode().to_frame() for col_name in columns[1:]
+            native_frame[col_name].explode().to_frame(col_name)
```

Comment:
In principle this is needed only in case a dataframe column name is None. Probably rare, but at this point I stopped asking what's possible when naming objects in pandas